ReflectionAnalyzers.Tests/TestHelper.cs

```diff
@@ -0,0 +1,29 @@
+namespace ReflectionAnalyzers.Tests
+{
+    using System.IO;
+    using Gu.Roslyn.Asserts;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using NUnit.Framework;
+
+    public static class TestHelper
+    {
+        public static MetadataReference CompileBinaryReference(string code)
+        {
+            var binaryReferencedCompilation = CSharpCompilation.Create(
+                "BinaryReferencedAssembly",
+                new[] { CSharpSyntaxTree.ParseText(code) },
+                AnalyzerAssert.MetadataReferences,
+                new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+
+            using (var binaryReferencedContent = new MemoryStream())
+            {
+                var binaryEmitResult = binaryReferencedCompilation.Emit(binaryReferencedContent);
+                Assert.That(binaryEmitResult.Diagnostics, Is.Empty);
```

Comment:
Should this be more lenient?

Reply:
Something you can figure out when it goers into the assert lib.

Reply:
Oh, missed that, makes sense to check there I think. Should help finding test bugs.

Reply:
Maybe you write the PR for the assert lib?

Reply:
Maaaybe...you'd have the best sense of how this fits into the big picture of that API, though.

Reply:
My preference these days is to not split into libs until the usage patterns have proven themselves in public projects.

Reply:
Maaaybe...you'd have the best sense of how this fits into the big picture of that API, though.

My preference these days is to not split into libs until the usage patterns have proven themselves in multiple projects.